TRAVIS

```diff
@@ -1,11 +1,41 @@
 
-Implement a test suite that can run both locally and on Travis CI, that verifies that the example code passes the test suite for each exercise.
+Implement a track test suite that can run both locally and on Travis CI. The track test suite should verify that each exercise makes sense, by running the exercise tests against the example solution.
 
-Things to consider:
+**Definition of terms**
 
-If some tests are marked as skipped or pending in some way, then CI should remove these annotations so that all the tests run against the example, not just the first test.
+- **exercise test suite**: the test suite that is delivered to Exercism users as part of an Exercism exercise
+- **track test suite**: the test suite that helps ensure that all of the exercise test suites in a language track are solvable
 
-If we need to rewrite the files in any way, we should copy the exercise to a tmp directory first, so that we don't accidentally end up committing changes that are just for continuous integration or testing purposes.
+**Background**
 
-We should make it possible to test just one exercise at a time, because testing everything can take a while, and very often people are just working on one exercise and will only need to run the full test suite before committing the change.
+When implementing an exercise test suite, we want to provide a good user experience for the people writing a solution to the exercise. People should not be confused or overwhelmed.
+
+In most Exercism language tracks, we simulate Test-Driven Development (TDD) by implementing the tests in order of increasing complexity. We try to ensure that each test either
+
+- helps triangulate a solution to be more generic, or
+- requires new functionality incrementally.
+
+Many test frameworks will randomize the order of the tests when running them. This is an excellent practice, which helps ensure that subsequent tests are not dependent on side effects from earlier tests. However, in order to simulate TDD we want tests to run *in the order that they are defined*, and we want them to *fail fast*, that is to say, as soon as the test suite encounters a failure, we want the execution to stop. This ensures that the person implementing the solution sees only one error or failure message at a time, unless they make a change which causes prior tests to fail.
+
+This is the same experience that they would get if they were implementing each new test themselves.
+
+Most testing frameworks do not have the necessary configuration options to get this behavior directly, but they often do have a way of marking tests as _skipped_ or _pending_. The mechanism for this will vary from language to language and from test framework to test framework.
+
+Whatever the mechanism—functions, methods, annotations, directives, or some other approach—these are text added to the test file. The person solving the exercise will need to edit the test file in order to "activate" each subsequent test.
```

Comment:
We've not run into it yet, but yeah, that might be necessary at some point. I'll update this.